spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/AddFilesProcedure.java

```diff
@@ -265,13 +283,13 @@ public String description() {
     return "AddFiles";
   }
 
-  private void validatePartitionSpec(Table table, Map<String, String> partitionFilter) {
+  private void validatePartitionFilter(Table table, Map<String, String> partitionFilter) {
     List<PartitionField> partitionFields = table.spec().fields();
     Set<String> partitionNames =
         table.spec().fields().stream().map(PartitionField::name).collect(Collectors.toSet());
```

Comment:
This is now wrong, we need to use the "compatible partition spec" we discovered instead of the current spec

Reply:
Fixed this to take partition spec and not always take the table latest spec.

Although, I had to make a refactor here, I moved the identity partition check from this function to another function so that I can check it at the beginning itself, because we can just exit early in that case.

Then, I am doing the `validatePartitionFilter` later, after we find a compatible spec (because now we have to pass it to this function)